### PR TITLE
Add `hostname` to `Process` entity

### PR DESCRIPTION
**Pull request by nupplaphil**
The process of background tasks are now bound to a specific `hostname`.
And we can now use the environment variable `NODE_ENV` to overwrite the `hostname` in case of cluster environments :-)

Another step to multi-daemon nodes :-)

I dropped the `host` table as well because I don't see any advantage and it's hard to cleanup this table because how can we determine if a `host` isn't available anymore .. I think this is too much complexity for too less advantages ...

Superseded #10601
Part of #9095

### CODE REVIEW

File: src/Core/Worker/Factory/Process.php [hunk 1]
```diff
@@ -8,28 +27,22 @@
 
 class Process extends BaseFactory implements ICanCreateFromTableRow
 {
+	public function determineHost(?string $hostname = null): string
+	{
+		if (empty($hostname)) {
+			$hostname = php_uname('n');
+		}
+
+		return strtolower($hostname);
```
Comment (MrPetovan) [inline src/Core/Worker/Factory/Process.php]: This is fine but what about a one liner?
```suggestion
		return strtolower($hostname ?? php_uname('n'));
```

File: src/Core/Worker/Repository/Process.php [hunk 2]
```diff
@@ -34,18 +34,25 @@
...
 
-	public function __construct(Database $database, LoggerInterface $logger, Factory\Process $factory)
+	/** @var string */
+	private $currentHost;
+
+	public function __construct(Database $database, LoggerInterface $logger, Factory\Process $factory, array $server)
 	{
 		parent::__construct($database, $logger, $factory);
+
+		$this->currentHost = $factory->determineHost($server[self::NODE_ENV] ?? '');
```
Comment (MrPetovan) [inline src/Core/Worker/Repository/Process.php]: Why empty string?
```suggestion
		$this->currentHost = $factory->determineHost($server[self::NODE_ENV] ?? null);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by annando** [comment]
Have you tried if the primary key really had been altered in the database? AFAIK the dbstructure mechanism can't handle changing primary keys. Also please add the "host" table to the list of tables that can be dropped in `DBStructure.php`.

**Comment by nupplaphil** [comment]
@annando This was the ouptut and I successfully updated from `1442` to `1443`:
```console
root@friendica:/vagrant# ./bin/console dbstructure update 
ALTER IGNORE TABLE `process` DROP INDEX `PRIMARY`, MODIFY `hostname` varchar(32) NOT NULL COMMENT 'The host name of the current node', ADD PRIMARY KEY(`pid`,`hostname`); 

root@friendica:/vagrant# ./bin/console dbstructure update 

root@friendica:/vagrant#
```

`host` is already included at `DBStructure::dropTables()`

[Edit] Got it: This works because neither `pid` nor `hostname` are auto-increments

**Review by MrPetovan — APPROVED** [review]

**Comment by annando** [comment]
@nupplaphil when changing the database structure, please execute the command `bin/console dbstructure dumpsql > database.sql` to update not only the `database.sql` file but also the documention. This is now done in #10962

**Pull request merged**